python/14-pdf-numeric-table.ipynb

```diff
@@ -0,0 +1,314 @@
+    "from IPython.display import Image, display, JSON, HTML\n",
+    "from textractcaller.t_call import call_textract, Textract_Features, call_textract_expense\n",
+    "from textractprettyprinter.t_pretty_print import convert_table_to_list\n",
+    "from trp import Document\n",
+    "import os\n",
+    "\n",
+    "\n",
+    "# variables\n",
+    "data_bucket = sagemaker.Session().default_bucket()\n",
+    "region = boto3.session.Session().region_name\n",
+    "account_id = boto3.client('sts').get_caller_identity().get('Account')\n",
+    "\n",
+    "os.environ[\"BUCKET\"] = data_bucket\n",
+    "os.environ[\"REGION\"] = region\n",
+    "role = sagemaker.get_execution_role()\n",
+    "\n",
+    "print(f\"SageMaker role is: {role}\\nDefault SageMaker Bucket: s3://{data_bucket}\")\n",
+    "\n",
+    "s3=boto3.client('s3')\n",
+    "textract = boto3.client('textract', region_name=region)\n",
+    "comprehend=boto3.client('comprehend', region_name=region)\n",
+    "comprehendmedical=boto3.client('comprehendmedical', region_name=region)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "file=\"./DemoTable.pdf\"\n",
+    "file_key=f\"idp/textract/demo/{os.path.basename(file)}\"\n",
+    "s3url=\"s3://\"+data_bucket+\"/\"+file_key\n",
+    "!aws s3 cp {file} {s3url} --only-show-errors"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "\n",
+    "from IPython.display import IFrame\n",
+    "IFrame(file, width=1500, height=600)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "Extract unstructured data with Amazon Textract \n",
+    "\n",
+    "Amazon Textract is an ML powered OCR service that is capable of detecting and extracting text from documents. Text data in the form of WORDS and LINES can be extracted from documents using Amazon Textract `DetectDocumentText` API. Let's extract the words and lines from the demonstration document."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true,
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Call Amazon Textract\n",
+    "response = call_textract(input_document=s3url)\n",
+    "\n",
+    "text=\"\"\n",
+    "# Print detected text\n",
+    "for item in response[\"Blocks\"]:\n",
+    "    if item[\"BlockType\"] == \"LINE\":\n",
+    "        text=text+\"\\n\"+item[\"Text\"]\n",
+    "print (text)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "# Detect Tables in raw text\n",
+    "\n",
+    "In this example using a threashold of 4 characters out of the 15 most used characters being digits works well to identify if the page is mostly constituted of numeric tables.\n",
+    "\n"
```

Comment:
explain in more detail how this implementation works. This is the key point of the notebook/sample, so we want to be very clear about this. Provide a second implementation as well, so the concept is understood.

Reply:
Added a second implementation focusing on each line.